internal/backend/db/dbgorm/ownership.go

```diff
@@ -120,40 +121,40 @@ func (gdb *gormdb) createEntityWithOwnership(
 	}
 
 	return gdb.transaction(ctx, func(tx *tx) error {
-		if err := tx.isUserEntity(user, idsToVerifyOwnership...); err != nil {
+		if err := tx.isUserEntity(uid, idsToVerifyOwnership...); err != nil {
 			return err
 		}
-		if err := create(tx.db, user); err != nil { // create
+		if err := create(tx.db, uid); err != nil { // create
 			return err
 		}
-		return saveOwnershipForEntities(tx.db, user, ownerships...) // ensure user and add ownerships
+		return saveOwnershipForEntities(tx.db, uid, ownerships...) // ensure user and add ownerships
 	})
 }
 
-func isUserEntity(db *gorm.DB, uid string, ids ...sdktypes.UUID) error {
+func EnsureUserAccessToEntities(db *gorm.DB, uid string, ids ...sdktypes.UUID) ([]scheme.Ownership, error) {
 	if len(ids) == 0 {
-		return nil
+		return nil, nil
 	}
 
 	var oo []scheme.Ownership
-	if err := db.Model(&scheme.Ownership{}).Where("entity_id IN ?", ids).Select("user_id").Find(&oo).Error; err != nil {
-		return err
+	if err := db.Model(&scheme.Ownership{}).Where("entity_id IN ?", ids).Select("user_id", "entity_type").Find(&oo).Error; err != nil {
+		return nil, err
 	}
 
 	if len(oo) < len(ids) {
-		return gorm.ErrRecordNotFound
+		return oo, gorm.ErrRecordNotFound
```

Comment:
what is oo?

Reply:
`o` is ownerships, `oo` is ownershipS :) , e.g. many

Reply:
Maybe `ows`

Reply:
please change to something meaningful like ownerships

Reply:
fixed